src/main/include/log4cxx/defaultconfigurator.h

```diff
@@ -39,19 +35,39 @@ class LOG4CXX_EXPORT DefaultConfigurator
 
 	public:
 		/**
-		Add a ConsoleAppender that uses PatternLayout
-		using the PatternLayout#TTCC_CONVERSION_PATTERN and
-		prints to <code>stdout</code> to the root logger.*/
-		static void configure(log4cxx::spi::LoggerRepositoryPtr repository);
+		Configure \c repository.
+
+		If the configuration file name has not been provided by a call to setConfigurationFileName(),
+		the environment variables 'LOG4CXX_CONFIGURATION' and 'log4j.configuration' are examined.
+
+		If the configuration file name has not been provided by any of the previous approaches
+		the current directory is examined for a file named either
+		"log4cxx.xml", "log4cxx.properties", "log4j.xml" or "log4j.properties".
```

Comment:
what happens if this fails?  Does it still fall back to the old way of using the `PatternLayout`?

Reply:
You will see the unhelpful "Please initialize the log4cxx system properly".

It also writes a LogLog::debug message.